test/teiserver/o_auth/application_query_test.exs

```diff
@@ -108,4 +108,82 @@ defmodule Teiserver.OAuth.ApplicationQueryTest do
                ApplicationQueries.get_stats([app.id, other_app.id])
     end
   end
+
+  describe "user application management" do
+    setup [:setup_app]
+
+    test "list_authorized_applications returns correct apps", %{user: user, app: app} do
+      # Create tokens for the app
+      OAuthFixtures.token_attrs(user.id, app) |> OAuthFixtures.create_token()
+
+      apps = ApplicationQueries.list_authorized_applications(user.id)
+
+      assert length(apps) == 1
+      assert hd(apps).id == app.id
```

Comment:
You can also do these two tests using pattern matching: `[%{id: ^app_id}] = apps` or, if you prefer not to have a `app_id` var there only for pattern match, you can get the app like so: `[app] = apps`.

I find that you get nicer error messages when something goes wrong this way. I also find that more intuitive but this is more subjective.